main.go

```diff
@@ -293,12 +317,11 @@ func main() {
 	r.POST("/", func(context *gin.Context) {
 		payload, err := github.ValidatePayload(context.Request, conf.githubSecret)
 		if err != nil {
-			log.Warnln("payload failed to validate, ignoring.")
+			logrus.Warnln("payload failed to validate, ignoring.")
 			return
 		}
 
-		// Set unique ID for the logger
-		log.AddHook(newDeliveryFieldHook(github.DeliveryID(context.Request)))
+		context.Set("delivery", github.DeliveryID(context.Request))
 
 		go processGitHubWebhook(context, payload, githubClient, conf)
```

Comment:
The request handler is already running in a separate go routine.
```suggestion
		processGitHubWebhook(context, payload, githubClient, conf)
```

Reply:
Maybe, but what we are trying to achieve here is to reply to GitHub before the full processing of the request.

This was actually a big problem before, where many of the requests were timing out (from GitHub point of view) and we could see in `gin` logs that we only responded after all the processing was done.

Reply:
Ahh, OK. Makes sense.